src/main/kotlin/gg/flyte/christmas/minigame/games/Spleef.kt

```diff
@@ -203,54 +352,324 @@ class Spleef : EventMiniGame(GameConfig.SPLEEF) {
         }
 
         listeners += event<PlayerToggleFlightEvent> {
-            if (player.gameMode != GameMode.SURVIVAL) return@event // elimed players alr in ADVENTURE
+            if (player.gameMode != GameMode.SURVIVAL) return@event
             isCancelled = true
 
-            if (doubleJumps[player]!! > 0) {
-                player.isFlying = false
-                player.velocity = player.location.direction.multiply(0.5).add(Vector(0.0, 1.25, 0.0))
-                player.playSound(Sound.ITEM_TOTEM_USE)
-                doubleJumps[player] = doubleJumps[player]!! - 1
-            } else player.allowFlight = false
+            if (unlimitedJumps) {
+                doubleJump(player)
+            } else if (doubleJumps[player.uniqueId]!! > 0) {
+                doubleJump(player)
+
+                doubleJumps[player.uniqueId] = doubleJumps[player.uniqueId]!! - 1
+            } else {
+                player.allowFlight = false
+            }
+        }
+
+        listeners += event<ProjectileLaunchEvent> {
+            if (entity !is Snowball) return@event
+
+            if (powerfulSnowballs) {
+                entity.velocity = entity.velocity.multiply(2)
+            }
         }
 
         listeners += event<ProjectileHitEvent> {
+            if (hitEntity != null && entity.shooter !is Player) {
```

Comment:
I dont exactly know what this means, but I think you should change to `is SnowGolem` because atm this is a bit confusing,

Reply:
The point is that snowballs from snowball rain should also phase through entities. It used to be is SnowGolem before I added snowball rain